server/src/main/java/org/elasticsearch/index/mapper/CompletionFieldMapper.java

```diff
@@ -436,8 +436,9 @@ public void parse(ParseContext context) throws IOException {
         Token token = parser.currentToken();
         Map<String, CompletionInputMetaData> inputMap = new HashMap<>(1);
 
-        // ignore null values
-        if (token == Token.VALUE_NULL) {
+        if (context.externalValueSet()) {
+            inputMap = (Map<String, CompletionInputMetaData>) context.externalValue();
```

Comment:
The external value is not always set by another `completion` mapper so you need to check the type.
I think it would be simpler to just add an input field in the `CompletionInputMetaData` class and do something like:
````
 if (context.externalValueSet()) {
   if (context.externalValue().getClass().equals(CompletionInputMetaData.class)) {
     CompletionInputMetaData inputAndMeta = (CompletionInputMetaData) context.externalValue();
     inputMap = Collections.singletonMap(inputAndMeta.input(), inputAndMeta);
   } else {
     inputMap = Collections.singletonMap(context.externalValue().toString(), new CompletionInputMetaData(Collections.<String, Set<CharSequence>>emptyMap(), 1));
  }...
````

Reply:
well spotted, indeed missed these scenarios where external value is set by GeoPointFIeldMapper. Will add test cases and cover that.

Also, good suggestion. Will extend CompletionInputMetaData to include inputField and return that as a toString